pkg/reconciler/function/semantic/semantic.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright (c) 2020 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semantic
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// Semantic can do semantic deep equality checks for Kubernetes API objects.
+//
+// For a given comparison function
+//
+//   comp(a, b interface{})
+//
+// 'a' should always be the desired state, and 'b' the current state for
+// DeepDerivative comparisons to work as expected.
+var Semantic = conversion.EqualitiesOrDie(
+	deploymentEqual,
```

Comment:
`deploymentEqual` can probably be removed from this repo since it isn't used. I don't mind if you prefer to keep it though.